cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -56,4 +62,71 @@ public int serviceTicketCount() {
                 this.getClass().getName(), Integer.MIN_VALUE);
       return Integer.MIN_VALUE;
     }
+
+    @Override
+    public final boolean deleteTicket(final String ticketId) {
+        if (ticketId == null) {
+            return false;
+        }
+
+        final Ticket ticket = getTicket(ticketId);
+        if (ticket == null) {
+            return false;
+        }
+
+        if (ticket instanceof TicketGrantingTicket) {
+            logger.debug("Removing children of ticket [{}] from the registry.", ticket.getId());
+            final TicketGrantingTicket tgt = (TicketGrantingTicket) ticket;
+            deleteChildren(tgt);
+
+            final Collection<ProxyGrantingTicket> proxyGrantingTickets = tgt.getProxyGrantingTickets();
+            for (final ProxyGrantingTicket proxyGrantingTicket : proxyGrantingTickets) {
+                logger.debug("Removing proxy-granting ticket [{}]", proxyGrantingTicket.getId());
+                deleteTicket(proxyGrantingTicket.getId());
+            }
+
+        }
+        logger.debug("Removing ticket [{}] from the registry.", ticket);
+        return deleteSingleTicket(ticketId);
```

Comment:
No reverse order does not work specially in JPA cases because you have to maintain referential integrity. You always want to start with the children, and the move up. You can't leave orphan objects around, if you start by killing parents first. 

I call this the "cabin pressure" strategy. "In the case of a drop in cabin pressure, always assist your children first by helping them put the oxygen mask on, and then put on yours" :)